crates/democracy/src/lib.rs

```diff
@@ -123,68 +112,37 @@ pub type PropIndex = u32;
 pub type ReferendumIndex = u32;
 
 type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
-type NegativeImbalanceOf<T> =
-    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
-
-#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
-pub enum PreimageStatus<AccountId, Balance, BlockNumber> {
-    /// The preimage is imminently needed at the argument.
-    Missing(BlockNumber),
-    /// The preimage is available.
-    Available {
-        data: Vec<u8>,
-        provider: AccountId,
-        deposit: Balance,
-        since: BlockNumber,
-        /// None if it's not imminent.
-        expiry: Option<BlockNumber>,
-    },
-}
-
-impl<AccountId, Balance, BlockNumber> PreimageStatus<AccountId, Balance, BlockNumber> {
-    fn to_missing_expiry(self) -> Option<BlockNumber> {
-        match self {
-            PreimageStatus::Missing(expiry) => Some(expiry),
-            _ => None,
-        }
-    }
-}
-
-// A value placed in storage that represents the current version of the Democracy storage.
-// This value is used by the `on_runtime_upgrade` logic to determine whether we run
-// storage migration logic.
-#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
-enum Releases {
-    V1,
-}
+pub type CallOf<T> = <T as frame_system::Config>::RuntimeCall;
+pub type BoundedCallOf<T> = Bounded<CallOf<T>>;
```

Comment:
What is the bound? Can we ever run into a size limit?

Reply:
Actually it is good that you pointed this out, I was under the (incorrect) assumption that `Bounded` was simply an encoded call with a size limit but that seems not to be the case: https://github.com/paritytech/substrate/blob/f59f5019333c44b2513852f7ddf30514430ec78f/frame/support/src/traits/preimages.rs#L44

Reply:
But this will not actually resolve https://github.com/interlay/interbtc/issues/951